dbms/src/AggregateFunctions/AggregateFunctionUniqCombined.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <Common/CombinedCardinalityEstimator.h>
+
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeUUID.h>
+#include <DataTypes/DataTypesNumber.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <AggregateFunctions/UniqCombinedBiasData.h>
+#include <AggregateFunctions/UniqVariadicHash.h>
+
+#include <ext/bit_cast.h>
+
+#include <Columns/ColumnVector.h>
+#include <Columns/ColumnsNumber.h>
+#include <Common/FieldVisitors.h>
+#include <Common/SipHash.h>
+
+#include <Common/typeid_cast.h>
+
+namespace DB
+{
+namespace detail
+{
+    /** Hash function for uniqCombined.
+     */
+    template <typename T>
+    struct AggregateFunctionUniqCombinedTraits
+    {
+        static UInt32 hash(T x)
+        {
+            return static_cast<UInt32>(intHash64(x));
+        }
+    };
+
+    template <>
+    struct AggregateFunctionUniqCombinedTraits<UInt128>
+    {
+        static UInt32 hash(UInt128 x)
+        {
+            return sipHash64(x);
+        }
+    };
+
+    template <>
+    struct AggregateFunctionUniqCombinedTraits<Float32>
+    {
+        static UInt32 hash(Float32 x)
+        {
+            UInt64 res = ext::bit_cast<UInt64>(x);
+            return static_cast<UInt32>(intHash64(res));
+        }
+    };
+
+    template <>
+    struct AggregateFunctionUniqCombinedTraits<Float64>
+    {
+        static UInt32 hash(Float64 x)
+        {
+            UInt64 res = ext::bit_cast<UInt64>(x);
+            return static_cast<UInt32>(intHash64(res));
+        }
+    };
+
+} // namespace detail
+
+
+template <typename Key, UInt8 K>
+struct AggregateFunctionUniqCombinedDataWithKey
+{
+    using Set = CombinedCardinalityEstimator<Key,
+        HashSet<Key, TrivialHash, HashTableGrower<>>,
+        16,
+        K - 3,
```

Comment:
Missing comment.

Reply:
HyperLogLog uses 6 bits per cell.
HashSet uses `sizeof(Key) * 8 * 2` bits per element with maximum fill.

How `K - 3` is considered to be correct? Maybe it was already incorrect?